src/overlays/actors/ovl_Obj_Snowball/z_obj_snowball.c

```diff
@@ -28,80 +46,802 @@ const ActorInit Obj_Snowball_InitVars = {
+}
+
+void func_80B02D58(ObjSnowball* this, GlobalContext* globalCtx) {
+    s32 temp_v0 = func_800A8150(OBJSNOWBALL_GET_3F(&this->actor));
+
+    if (temp_v0 >= 0) {
+        Item_DropCollectible(globalCtx, &this->actor.home.pos, (OBJSNOWBALL_GET_7F00(&this->actor) << 8) | temp_v0);
+    }
+}
+
+void func_80B02DB0(ObjSnowball* this, GlobalContext* globalCtx) {
+    s32 pad;
+    ObjSnowballStruct2* ptr = &D_80B04F84[this->actor.home.rot.y];
+
+    Actor_SpawnAsChildAndCutscene(&globalCtx->actorCtx, globalCtx, ptr->unk_00, this->actor.home.pos.x,
+                                  this->actor.home.pos.y, this->actor.home.pos.z, this->actor.home.rot.x, 0,
+                                  this->actor.home.rot.z, this->actor.params | ptr->unk_02,
+                                  ActorCutscene_GetAdditionalCutscene(this->actor.cutscene), this->actor.unk20, NULL);
+}
+
+void func_80B02E54(ObjSnowball* this, GlobalContext* globalCtx) {
+    ObjSnowballStruct2* ptr = &D_80B04F84[this->actor.home.rot.y];
+
+    Actor_SpawnAsChildAndCutscene(&globalCtx->actorCtx, globalCtx, ptr->unk_00, this->actor.home.pos.x,
+                                  this->actor.home.pos.y, this->actor.home.pos.z, this->actor.home.rot.x, 0,
+                                  this->actor.home.rot.z, this->actor.params | ptr->unk_02, -1, this->actor.unk20,
+                                  NULL);
+}
+
+void func_80B02EE4(ObjSnowball* this, GlobalContext* globalCtx) {
+    Vec3f spAC;
+    Vec3f spA0;
+    Vec3f sp94;
+    Vec3s* hitPos = &this->collider.elements->info.bumper.hitPos;
+    s32 i;
+
+    for (i = 0; i < 4; i++) {
+        sp94.x = ((Rand_ZeroOne() * 14.0f) - 7.0f) + hitPos->x;
+        sp94.y = ((Rand_ZeroOne() * 14.0f) - 7.0f) + hitPos->y;
+        sp94.z = ((Rand_ZeroOne() * 14.0f) - 7.0f) + hitPos->z;
+
+        spA0.x = (Rand_ZeroOne() - 0.5f) * 1.6f;
+        spA0.y = -0.8f;
+        spA0.z = (Rand_ZeroOne() - 0.5f) * 1.6f;
+
+        spAC.x = spA0.x * -0.06f;
+        spAC.y = spA0.y * -0.06f;
+        spAC.z = spA0.z * -0.06f;
+
+        func_800B0E48(globalCtx, &sp94, &spA0, &spAC, &D_80B04FB4, &D_80B04FB8, (s32)(Rand_ZeroOne() * 30.0f) + 15,
+                      (s32)(Rand_ZeroOne() * 40.0f) + 30);
+    }
+}
+
+void func_80B030F8(ObjSnowball* this, GlobalContext* globalCtx) {
+    s32 pad;
+    f32 temp_f28 = sqrtf(this->unk_20C);
+    Vec3f spFC;
+    Vec3f spF0;
+    Vec3f spE4;
+    s32 i;
+    s32 phi_s6;
+    Gfx* temp_s2;
+    s16 scale;
+    s32 temp_s1;
+    f32 temp_f20;
+    f32 temp_f22;
+    f32 temp_f26;
+    s16 temp_s0;
+    s32 temp_s7;
+    f32 phi_f22;
+    s32 gravity;
+    s32 phi_s0;
+    s32 phi_s4;
+
+    for (i = 0, phi_s6 = 0; i < 25; i++, phi_s6 += 0xA3D) {
+        temp_s7 = i & 7;
+        if (temp_s7 < 4) {
+            temp_s2 = D_80B04FC8[2];
+            gravity = -280;
+            phi_s4 = 0;
+            phi_s0 = 0x40;
+            phi_f22 = 1.0f;
+        } else {
+            if (temp_s7 < 6) {
```

Comment:
You could mix this `if` condition
```suggestion
        } else if (temp_s7 < 6) {
```